snapcraft/internal/deltas/_deltas.py

```diff
@@ -84,6 +87,15 @@ def _check_delta_gen_tool(self):
         if not file_utils.executable_exists(self.delta_tool_path):
             raise DeltaToolError(delta_tool=self.delta_tool_path)
 
+    def _check_delta_size_constraint(self, delta_path):
+        """Ensure delta is sufficiently smaller than target snap."""
```

Comment:
Can you explain this a bit? This will cause the full snap to be pushed instead, when it seems the delta would still result in bandwidth savings. 90% seems a bit arbitrary, I'd like to see some comments addressing that.